tests/test_utils/test_model_cache.py

```diff
@@ -67,10 +67,15 @@ def test_set_cpu_full():
 
 @pytest.mark.skipif(get_device() == "cpu", reason="GPU not available")
 def test_set_gpu_full():
+    device = get_device()
     cache = GPUModelCache(
-        max_cpu_memory_gb=1, max_gpu_memory_gb=0.0000001, device=get_device()
+        max_cpu_memory_gb=1, max_gpu_memory_gb=0.0000001, device=device
     )
-    assert cache.max_cpu_memory == 1073741824
+    if device is None:
```

Comment:
get_device never returns `None` so I'm not sure what this is handling

Reply:
Looking at this again. You are correct. Trying to figure out why I'm getting max_cpu_memory = 0 on my M1 mac.

The test isn't run when the device is "cpu". Should it also not run when the device is mps?

Reply:
<img width="989" alt="Screenshot 2023-09-18 at 11 23 49 PM" src="https://github.com/brycedrennan/imaginAIry/assets/31525633/6fd6516a-a42b-49b8-9641-037272fd49b6">

Reply:
max_cpu_memory is set to 0 if run with cpu or mps so I made an if statement here for if you're running on a mac.